src/open_inwoner/accounts/views/profile.py

```diff
@@ -343,3 +348,30 @@ def form_valid(self, form):
             self.request.user, _("users newsletter subscriptions were modified")
         )
         return HttpResponseRedirect(self.get_success_url())
+
+
+class MyAppointmentsView(
+    LogMixin, LoginRequiredMixin, CommonPageMixin, BaseBreadcrumbMixin, TemplateView
+):
+    template_name = "pages/profile/appointments.html"
+
+    def get_context_data(self, **kwargs) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+        # TODO email should be verified
+        try:
+            client = QmaticClient()
+        except NoServiceConfigured:
+            logger.exception("Error occurred while creating Qmatic client")
+            context["appointments"] = []
+        else:
+            context["appointments"] = client.list_appointments_for_customer(
+                quote(self.request.user.email)
```

Comment:
This assumes that the municipality uses a citizen's email as `external_id` in Qmatic (still awaiting confirmation from Hoorn if this is correct). If we want to support `bsn` or something else, we'll probably have to add a feature flag or choicefield on the config to enable that.

Reply:
This `quote()` doesn't belong here but should be in the client method. It is not the responsibility of the using code to encode the values (as conceptually it doesn't even know how the client works internally).

This relates to my earlier comment I put in the client.